### ISSUE TITLE
Concatenating outputs from convolutional layers produces an error

### ISSUE THREAD

**Issue by goobur** [plaidml/plaidml#278]
It seems like the code generation for the Concatenate layer is broken. I have a specific network that I've been working on, but a more minimal example of code that should work (but breaks) is here: 
https://towardsdatascience.com/fingerprint-denoising-and-inpainting-using-fully-convolutional-networks-e24714c3233
or directly to the code: 
https://github.com/CVxTz/fingerprint_denoising/blob/master/code/baseline_aug.py

**Comment by brianretford** [comment]
Can you copy the exact error you're seeing just to make sure we're reproducing the same issue?

**Comment by iperov** [comment]
use `keras.layers.Concatenate (axis=-1) ( [ tensors...] )`

**Comment by goobur** [comment]
@brianretford Sure thing, the error I'm getting is "Error when checking target: expected dropout_19 to have shape (256, 256, 1) but got array with shape (256, 256, 3)" 

@iperov Even if I use axis=-1, I still get the same error. Regardless, even if that were to work, this is still a bug -- I my tensors match along the dimensions that matter and this /should/ work by that logic.

**Comment by SleepProgger** [comment]
That error has nothing to do with the concatenate layer.
Its just telling you that the validation data you feed in has 3 depth channel (256,256,3) while you models output has 1 (256,256,1).

*Off topic, but might be of interest for you: https://towardsdatascience.com/dont-use-dropout-in-convolutional-networks-81486c823c16?gi=639f21200b3*

**Comment by goobur** [comment]
Ah, ddang. Yes, that was the issue with the Unet -- I have RGB data, but forgot to change the Unet to that output size.

Also, on closer inspection, the bug was elsewhere in how I was building my model -- I got it all debugged now. My bad :)

@SleepProgger That was an interesting read, thanks!
